server/src/main/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapper.java

```diff
@@ -1052,6 +1070,16 @@ public String toString() {
         boolean supportsElementType(ElementType elementType) {
             return elementType != ElementType.BYTE;
         }
+
+        @Override
+        boolean updatableTo(IndexOptions update) {
+            boolean sameType = update.type.equals(this.type);
```

Comment:
Could you disable updates to `confidence_interval` as well? Quantization could not behave as expected if they update the confidence_interval and expect different quantiles to be created.